lib/spack/spack/build_environment.py

```diff
@@ -59,6 +59,11 @@
 SPACK_DEBUG_LOG_DIR    = 'SPACK_DEBUG_LOG_DIR'
 
 
+# Platform-specific library suffix.
+dso_suffix = 'dylib' if sys.platform == 'darwin' else 'so'
```

Comment:
I figured it would be popular :). It should likely move to be an attribute of the platform when we get the newarch PR merged.